### PR TITLE
Revert liveness rule only valid for the same finalizer generation changes

### PR DESCRIPTION
**Pull request by linh2931**
The change to the liveness rule (https://github.com/AntelopeIO/spring/issues/293) create scenarios in which liveness is lost/reduced. Revert the changes. The proper safety solution will be handled by other upcoming PRs.

Resolves https://github.com/AntelopeIO/spring/issues/375

### CODE REVIEW

**Review by heifner — APPROVED** [review]

File: libraries/chain/finality/finalizer.cpp [hunk 1]
```diff
@@ -99,7 +98,7 @@ bool finalizer::maybe_update_fsi(const block_state_ptr& bsp) {
    auto& final_on_strong_qc_block_ref = bsp->core.get_block_reference(bsp->core.final_on_strong_qc_block_num);
    if (final_on_strong_qc_block_ref.timestamp > fsi.lock.timestamp && bsp->timestamp() > fsi.last_vote.timestamp) {
       fsi.lock = final_on_strong_qc_block_ref;
```
Comment (greg7mdp) [inline libraries/chain/finality/finalizer.cpp:100]: I believe we should set the timestamp on `fsi.lock`. Also on line 83.

**Review by greg7mdp — CHANGES_REQUESTED** [review]

File: libraries/chain/finality/finalizer.cpp (hunk 1 above)
Comment (linh2931) [reply libraries/chain/finality/finalizer.cpp:100]: `fsi.lock` is from `final_on_strong_qc_block_ref` which already contains the timestamp. What other timestamp to set?

Comment (greg7mdp) [reply libraries/chain/finality/finalizer.cpp:100]: Oh, my bad, sorry!

**Review by greg7mdp — APPROVED** [review]

**Comment by ericpassmore** [comment]
Note:start
group: STABILITY
category: INTERNALS
summary: Revert previous changes to liveness rule. After evaluation the previous changes reduced liveness, and engineering identified a better solution.
Note:end

**Pull request merged**